Flutter_Weather/lib/models/weather_extension.dart

```diff
@@ -54,22 +136,105 @@ class DayWeather{
     return DateFormatter.weekDay(initDate.add(Duration(hours: weathers.first.timepoint)));
   }
 
-  WeatherData ? get weatherNow{
-    try {
-      DateTime now = DateTime.now();
-      return weathers.firstWhere((weatherData) =>
-        now.difference(initDate.add(Duration(hours: weathers.first.timepoint))).inMinutes <= 90);
-    } on Exception catch (e) {
-      Logger.logError(
-      className: "Day Weather",
-      methodName: "get WeatherNow",
-      message: e.toString());
-    }
+  WeatherData get weatherNow{
+    DateTime now = DateTime.now().toUtc();
+    WeatherData weatherNow = weathers.firstWhere((weatherData) =>
+      now.difference(initDate.add(Duration(hours: weatherData.timepoint)).toUtc()).abs().inMinutes <= 90);
+    print(now);
+    DateTime chosenDate = initDate.add(Duration(hours: weatherNow.timepoint));
+    print(chosenDate);
+    print(now.difference(chosenDate).abs().inMinutes);
+    return weatherNow;
+  }
+
+  WeatherData get typicalWeather{
+    WeatherData typicalWeather = weathers.first;
+    int max = 0;
+    final Map<String, List<WeatherData>> allWeathers =
+    groupBy(weathers, (WeatherData weather) {
+      return weather.weather;
+    });
+    allWeathers.forEach((key, value) {
+      if (value.length > max){
+        max = value.length;
+        typicalWeather = value.first;
+      }
+    });
+    return typicalWeather;
   }
 
   int get upperLimitTemp
     => weathers.map((hourData) => hourData.temp2m).toList().reduce(max);
 
   int get lowerLimitTemp
     => weathers.map((hourData) => hourData.temp2m).toList().reduce(min);
+  
+  List<WeatherData> get todayWeathers{
+    return weathers.where((weather) => initDate.add(Duration(hours: weather.timepoint)).hour <15).toList(); 
+  }
+  
+  WeatherData get todayTypicalWeather{
```

Comment:
todayTypicalWeather should be the WeatherData with the most occurrences of the day.
(group `todayWeathers ` by `weather` property and pick the group with the largest number.)